lib/dev_repo.ml

```diff
@@ -7,25 +7,27 @@ let from_string s = s
 let to_string t = t
 let pp = Fmt.string
 
-let rec repeat_while_some x ~f =
-  match f x with None -> x | Some x -> repeat_while_some x ~f
-
-(* Attempt to split a string by calling [split s], then choose a side of the
-   result with [side], returning [s] if a split is not possible *)
-let try_split_side s ~(split : string -> (string * string) option) ~side =
-  Base.Option.value_map (split s) ~f:side ~default:s
+let repo_name_regexp =
+  let open Re in
+  compile
+    (seq [
+        char '/';
+        rep (char '.');
+        group (rep (compl [ char '.'; char '/' ]));
+        char '.';
+        rep (compl [ char '/' ]);
+        rep (char '/');
+        stop
+      ])
 
 let repo_name t =
-  Uri.of_string t |> Uri.path
-  |> repeat_while_some ~f:(Base.String.chop_suffix ~suffix:"/")
-  |> try_split_side ~split:(Base.String.rsplit2 ~on:'/') ~side:snd
-  |> repeat_while_some ~f:(Base.String.chop_prefix ~prefix:".")
-  |> try_split_side ~split:(Base.String.lsplit2 ~on:'.') ~side:fst
-  |> function
-  | "" ->
-      Rresult.R.error_msgf
-        "unexpected empty string while computing name for dev_repo: \"%s\"" t
-  | non_empty -> Ok non_empty
+  let path = Uri.of_string t |> Uri.path in
+  match Re.exec_opt repo_name_regexp path with
+  | Some group ->
+    Ok (Re.Group.get group 0)
```

Comment:
Thanks @reynir for the PR!

It seems this regexp is not equivalent to the previous code, as some of the tests are failing. For example, `repo_name` returns an error on `git://github.com/ocamllabs/opam-monorepo` instead of `Ok "opam-monorepo"`.

Could you revisit it and make sure the tests are passing locally? (unfortunately the CI is not in a working state)

Reply:
I fixed the regular expression and also my usage of `Re.Group`. Two tests are failing due to the ordering of packages not being the same. I will look into that as well.

Reply:
Now fixed. The fix is a good argument for factoring out the `Base.Result.all` replacement.